compiler/rustc_lint/src/types.rs

```diff
@@ -727,7 +730,33 @@ declare_lint! {
     "proper use of libc types in foreign item definitions"
 }
 
-declare_lint_pass!(ImproperCTypesDefinitions => [IMPROPER_CTYPES_DEFINITIONS]);
+declare_lint! {
+    /// The AIX target follows the power alignment rule. This rule specifies
+    /// that structs with either a:
+    ///   - floating-point data type as its first member, or
+    ///   - first member being an aggregate whose recursively first member is a
+    ///     floating-point data type
+    /// must have its natural alignment. This is currently unimplemented within
+    /// the compiler, so a warning is produced in these situations.
+    ///
+    /// ### Example
+    ///
+    /// ```rust
+    /// pub struct Floats {
+    ///     a: f64,
+    ///     b: u8,
+    ///     c: f64,
```

Comment:
Please give what the layout for this struct "should be" according to AIX's alignment rules... it is supposedly `offset_of!(Floats, b) == 8` and `offset_of!(Floats, c) == 12)`, according to the description given. That means only `c` deviates, currently.

Reply:
Thank you for the suggestion. I have added the suggestion of adding the offsets of the struct fields.

Reply:
Thank you for writing that down.